src/vcpkg/vcpkgpaths.cpp

```diff
@@ -355,60 +355,53 @@ namespace vcpkg
 
     static Path lockfile_path(const VcpkgPaths& p) { return p.vcpkg_dir() / "vcpkg-lock.json"; }
 
-    VcpkgPaths::VcpkgPaths(Filesystem& filesystem, const VcpkgCmdArguments& args)
-        : m_pimpl(std::make_unique<details::VcpkgPathsImpl>(
-              filesystem,
-              args.feature_flag_settings(),
-              Util::Enum::to_enum<ToolCache::RequireExactVersions>(args.exact_abi_tools_versions.value_or(false))))
+    static Path determine_root(const Filesystem& fs, const Path& original_cwd, const VcpkgCmdArguments& args)
     {
-        original_cwd = filesystem.current_path(VCPKG_LINE_INFO);
-#if defined(_WIN32)
-        original_cwd = vcpkg::win32_fix_path_case(original_cwd);
-#endif // _WIN32
-
         if (args.vcpkg_root_dir)
         {
-            root = filesystem.almost_canonical(*args.vcpkg_root_dir, VCPKG_LINE_INFO);
+            return fs.almost_canonical(*args.vcpkg_root_dir, VCPKG_LINE_INFO);
         }
         else
         {
-            root = filesystem.find_file_recursively_up(original_cwd, ".vcpkg-root", VCPKG_LINE_INFO);
-            if (root.empty())
+            auto ret = fs.find_file_recursively_up(original_cwd, ".vcpkg-root", VCPKG_LINE_INFO);
+            if (ret.empty())
             {
-                root = filesystem.find_file_recursively_up(
-                    filesystem.almost_canonical(get_exe_path_of_current_process(), VCPKG_LINE_INFO),
+                return fs.find_file_recursively_up(
+                    fs.almost_canonical(get_exe_path_of_current_process(), VCPKG_LINE_INFO),
                     ".vcpkg-root",
                     VCPKG_LINE_INFO);
             }
+            return ret;
         }
+    }
 
-        Checks::check_exit(VCPKG_LINE_INFO, !root.empty(), "Error: Could not detect vcpkg-root.");
-        Debug::print("Using vcpkg-root: ", root, '\n');
-        const auto vcpkg_root_file = root / ".vcpkg-root";
-        auto bundle_file = filesystem.read_contents(vcpkg_root_file, VCPKG_LINE_INFO);
-        if (!std::all_of(bundle_file.begin(), bundle_file.end(), Parse::ParserBase::is_whitespace))
+    static void load_bundle_file(const Filesystem& fs, const Path& root, details::VcpkgPathsImpl& impl)
+    {
+        const auto vcpkg_bundle_file = root / "vcpkg-bundle.json";
+        if (fs.exists(vcpkg_bundle_file, IgnoreErrors{}))
         {
-            auto maybe_bundle_doc = Json::parse(bundle_file, vcpkg_root_file.native());
+            auto bundle_file = fs.read_contents(vcpkg_bundle_file, VCPKG_LINE_INFO);
```

Comment:
This should do 

```
auto bundle_file = fs.read_contents(vcpkg_bundle_file, IgnoreErrors{});
if (!bundle_file.empty()) {
}
```

rather than checking exists first.